selfdrive/car/tests/test_models.py

```diff
@@ -251,5 +257,33 @@ def test_panda_safety_carstate(self):
     self.assertFalse(len(failed_checks), f"panda safety doesn't agree with openpilot: {failed_checks}")
 
 
+def load_tests(_routes, ci=True):
+  test_suite = unittest.TestSuite()
+  for car_model, test_route in _routes:
+    test_case_args = {"car_model": car_model, "ci": ci}
+    if test_route is not None:
+      test_case_args.update({"route": test_route.route, "segment": test_route.segment})
+
+    # create new test case and discover tests
+    CarModelTestCase = type("CarModelTestCase", (TestCarModel,), test_case_args)
```

Comment:
There seems to be no support I could find to create multiple test cases from one class with different arguments very easily. Creating dynamic subclasses works and was the simplest method I found.